### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by leem42**
Added new styling for publications card
Added Table of Contents Widget
Added test for Table of Contents

### CODE REVIEW

File: src/lib/containers/MarkdownSynapse.js [hunk 1]
```diff
@@ -233,13 +273,27 @@ class MarkdownSynapse extends React.Component {
...
             () => {
+                // replace all reference tags with id's that we can later target
                 let current = count++
                 return `<a href="" id="ref${current}">[${current}]</a>`
             }
         )
+
+        // for TOC
+        let tocHeadersRegex = /<h[1-6] toc="true">.*<\/h[1-6]>/gm
+        let tocId = "SRC-header-"
```
Comment (jay-hodgson) [inline src/lib/containers/MarkdownSynapse.js]: constant

File: src/lib/containers/MarkdownSynapse.js [hunk 2]
```diff
@@ -233,13 +273,27 @@ class MarkdownSynapse extends React.Component {
...
         let markup = this.createMarkup(this.state.text).__html.replace(/<span id="wikiReference.*?<span data-widgetparams.*?span>/g, 
             () => {
+                // replace all reference tags with id's that we can later target
                 let current = count++
                 return `<a href="" id="ref${current}">[${current}]</a>`
             }
         )
+
+        // for TOC
+        let tocHeadersRegex = /<h[1-6] toc="true">.*<\/h[1-6]>/gm
```
Comment (jay-hodgson) [inline src/lib/containers/MarkdownSynapse.js]: constant (reused a couple of times).  if pattern needs to be changed in the future, there should be one place that the dev needs to change it.

File: src/lib/containers/row_renderers/Publication.js [hunk 3]
```diff
@@ -26,50 +25,27 @@ class Publication extends React.Component {
...
-                <Utils.CardFooter>
-                    <div className="col-sm-2 hidden-xs">
+            <div className="SRC-portalCard SRC-typePublication SRC-layoutLandscape SRC-showMetadata">
+                <div className="SRC-cardThumbnail">
+                    <Utils.Icon type={PUBLICATION}/>
+                </div>
+                <div className="SRC-cardContent">
+                    <div className="SRC-type">Publication</div>
+                    <div className="SRC-title"><h3><a>{citation}</a></h3></div>
+                    <div className="SRC-author">Smita Bhatia / Jean L. Nakamuralue</div>
```
Comment (jay-hodgson) [inline src/lib/containers/row_renderers/Publication.js]: remove default author list

File: src/lib/containers/MarkdownSynapse.js (hunk 2 above)
Comment (leem42) [reply src/lib/containers/MarkdownSynapse.js]: the two regexes are slightly different, the second is of the form:

```js
/<h([1-6]) id="(.*)" .*toc="true">(.*)<\/h[1-6]>/gm
```

I can still make them constants though

**Pull request merged**